src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/SyntaxTreeTrim.qs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+// Trimmer Removes Unused Callables
+namespace Microsoft.Quantum.Testing.SyntaxTreeTrimming {
+    
+    @EntryPoint()
+    operation Main() : Unit {
+        UsedOp();
+    }
+
+    operation UsedOp() : Unit { UsedFunc(); }
+    operation UnusedOp() : Unit { UnusedFunc(); }
+
+    function UsedFunc() : Unit { }
+    function UnusedFunc() : Unit { }
+}
+
+// =================================
+
+// Trimmer Keeps UDTs
```

Comment:
Does the trimmer trim types as well, or does it only trim the constructors? If it is the latter I'd suggest to rename the test to "trimmer keeps udt constructors" to avoid confusion. Since constructors are intrinsic, I assume they are only trimmed if the flag to trim intrinsics is set to true? It would be good to have a test for both cases, i.e. to add a test for the case when keepAllIntrinsics is set to false.

Reply:
Thanks for pointing this out. I had thought that the constructors for the types would get bundled tightly with the UDTs themselves so that when I filtered for only the callables in the transformation, I would filter out the UDTs along with their constructors. The test worked as intended so I thought this was correct, but you've now pointed out that the tests was really only working by accident, with the constructors being body intrinsic and filtered out that way. The intent is to not ever filter out UDTs or their constructors, so I will need to update the transformation to do that, and update the tests to correctly test for that.

Reply:
Okay I've fixed the issue with the transformation, and updated the existing tests so that they won't accidently pass from this behavior. The keepAllIntrinsics sub-feature was turned off for all the tests that don't test for that, making for cleaner and stronger tests. I do believe that the keepAllIntrinsic sub-feature should be tests separately and independently, and the issue before was caused by having that feature intermixed with the other tests.
I've also updated the name here to reflect that both the UDTs and their constructors should always be kept.

Reply:
Ok, I am not sure I entirely follow what you are saying here, but looking at the tests, I see that there is a test ``Trimmer Keeps Intrinsics When Told`` that checks the functionality of the keepAllIntrinsics flag. So I think you are just saying that you prefer a separate test case. Sounds good.